examples/07_automatic_preprocessing_with_the_supervectorizer.py

```diff
@@ -0,0 +1,193 @@
+# Importing the data
+# ---------------
+# Let's fetch the dataset, and load X and y:
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from dirty_cat.datasets import fetch_employee_salaries
+
+pd.set_option('display.max_rows', None)
+pd.set_option('display.max_columns', None)
+pd.set_option('display.width', None)
+pd.set_option('display.max_colwidth', None)
+
+employee_salaries = fetch_employee_salaries()
+print(employee_salaries['DESCR'])
+
+X: pd.DataFrame = employee_salaries['data']
+y = employee_salaries['target']
+# We'll drop a few columns we don't want
+X.drop(
+    [
+        'Current Annual Salary',  # Too linked with target
+        'full_name',  # Not relevant to the analysis
+        '2016_gross_pay_received',  # Too linked with target
+        '2016_overtime_pay',  # Too linked with target
+        'date_first_hired'  # Redundant with "year_first_hired"
+    ],
+    axis=1,
+    inplace=True
+)
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.15, random_state=42
+)
+
+###############################################################################
+# Using the vectorizer
+# --------------------
+# Here is a simple workflow with the `SuperVectorizer` using a `Pipeline`:
+#
+# It's the typical and recommended way of using it.
+
+import numpy as np
+
+# For scikit-learn 0.24, we need to require the experimental feature
+from sklearn.experimental import enable_hist_gradient_boosting
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+from sklearn.pipeline import Pipeline
+
+from dirty_cat import SuperVectorizer
+
+
+pipeline = Pipeline([
+    ('vectorizer', SuperVectorizer(auto_cast=True)),
+    ('clf', HistGradientBoostingRegressor(random_state=42))
+])
+
+###############################################################################
+# For reference, let's perform a cross-validation
+
+from sklearn.model_selection import cross_val_score
+
+
+scores = cross_val_score(pipeline, X, y, scoring='r2')
+
+print(f'{scores=}')
+print(f'mean={np.mean(scores)}')
+print(f'std={np.std(scores)}')
+
+###############################################################################
+# Compared to
+# :ref:`example 02<sphx_glr_auto_examples_02_fit_predict_plot_employee_salaries.py>`,
+# the mean score is a pretty much the same as the all-|SE|.
+
+###############################################################################
+# Analyzing what it does
+# ----------------------
+# Let's perform the same workflow, but without the `Pipeline`, so we can
+# analyze its mechanisms along the way.
+
+from sklearn.ensemble import RandomForestRegressor
+
+
+sup_vec = SuperVectorizer(auto_cast=True)
+regressor = RandomForestRegressor(n_estimators=25, random_state=42)
+# Fit the SuperVectorizer
+X_train_enc = sup_vec.fit_transform(X_train, y_train)
+X_test_enc = sup_vec.transform(X_test)
+# And the regressor
+regressor.fit(X_train_enc, y_train)
```

Comment:
Maybe we should move this part after we have printed the feature names

Reply:
You mean for the RFR ?
To get the feature names, the SV must be fitted.